snippets/csharp/keywords/TrueFalseOperatorsExample.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+
+public struct LaunchStatus
+{
+    public static readonly LaunchStatus Green = new LaunchStatus(0);
+    public static readonly LaunchStatus Yellow = new LaunchStatus(1);
+    public static readonly LaunchStatus Red = new LaunchStatus(2);
+
+    private int status;
+
+    private LaunchStatus(int status)
+    {
+        this.status = status;
+    }
+
+    public static bool operator true(LaunchStatus x) => x == Green;
+    public static bool operator false(LaunchStatus x) => x == Red;
+
+    public static LaunchStatus operator &(LaunchStatus x, LaunchStatus y)
+    {
+        if (x == Red || y == Red || (x == Yellow && y == Yellow))
+        {
+            return Red;
+        }
+
+        return Green;
```

Comment:
I think there's an issue here. For `x & y & z` work correctly when `z` is `Yellow`, this should return `Yellow` when one of `x` or `y` is `Yellow`.  

Do you agree?

Reply:
@BillWagner 
> For `x & y & z` work correctly...

to clarify: by "correctly" you mean that the `&` overload must be associative, so `(x & y) & z == x & (y & z)`

Yes, I agree then. Will make an update soon.

Reply:
@BillWagner I've addressed your feedback; thanks for review.

I've checked the updated definition for all possible `x`, `y`, and `z` values: now the `&` operator is commutative and associative. In particular, that means that `x1 & ... & xN` is `Yellow` iff there is only one operand which is `Yellow` with all others being `Green`. That's why I've updated the `true` definition to return `true` value for `Yellow` operand. The following requirement is met:
> if only one sub-system is yellow, we can launch.